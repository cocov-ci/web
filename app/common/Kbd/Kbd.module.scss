@use "sass:color";
@import "styles/variables";

.base {
  background: $colorKbdBackground;
  border: 1px solid $colorKbdBorder;
  box-shadow:
    0 2px 3px 0 color.adjust($colorBlack, $alpha: -0.78),
    inset 0 3px 0 0 color.adjust($colorWhite, $alpha: -0.5);
  border-radius: 3px;
  padding: 5px;
  opacity: 0.59;
  font-family: var(--inconsolata-font);
  font-size: 15px;
  color: $colorBlack;
  letter-spacing: 0;
  text-align: center;
}
