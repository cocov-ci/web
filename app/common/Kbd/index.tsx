import classNames from 'classnames'
import React from 'react'

import styles from './Kbd.module.scss'

type BoxProps = {
  className?: string
  text: string
}

const Kbd = ({ className, text }: BoxProps) => {
  return <kbd className={classNames(styles.base, className)}>{text}</kbd>
}

export default Kbd
