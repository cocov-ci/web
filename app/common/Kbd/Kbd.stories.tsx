import type { Meta, StoryObj } from '@storybook/react'

import Kbd from './index'

const meta: Meta<typeof Kbd> = {
  title: 'Common/Kbd',
  component: Kbd,
  tags: ['autodocs'],
}

export default meta

type Story = StoryObj<typeof Kbd>

export const Default: Story = {
  args: { text: 'esc' },
}
