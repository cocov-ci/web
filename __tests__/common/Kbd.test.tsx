import { render, screen } from '@testing-library/react'

import Kbd from 'app/common/Kbd'

import '@testing-library/jest-dom'

describe('common/Kbd', () => {
  const KbdComponent = () => <Kbd text="esc" />

  it('renders Kbd without stats info', () => {
    render(<KbdComponent />)

    expect(screen.getByText('esc')).toBeVisible()
  })

  it('renders Kbd snapshots', () => {
    const { container } = render(<KbdComponent />)

    expect(container).toMatchSnapshot()
  })
})
