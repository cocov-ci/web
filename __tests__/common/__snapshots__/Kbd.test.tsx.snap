// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`common/Kbd renders Kbd snapshots 1`] = `
<div>
  <kbd
    class="base variable light"
  >
    esc
  </kbd>
</div>
`;
